src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Unix.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO
+{
+    public partial class FileSystemWatcher
+    {
+        protected static void UnlockRootDirectory(Microsoft.Win32.SafeHandles.SafeFileHandle handle)
+        {
+            Interop.Sys.FLock(handle, Interop.Sys.LockOperations.LOCK_UN);
```

Comment:
Consider adding an assert that this succeeds (but I agree we don't want to throw in release if it fails).